packages/core/src/model.ts

```diff
@@ -578,47 +595,97 @@ export class ChartModel {
 	/*
 	 * Fill scales
 	 */
-	protected setColorScale() {
-		const defaultColors = colorPalettes.DEFAULT;
-
+	protected setUserProvidedColorScale() {
 		const options = this.getOptions();
 		const userProvidedScale = Tools.getProperty(options, "color", "scale");
 
-		// If there is no valid user provided scale, use the default set of colors
 		if (
 			userProvidedScale === null ||
 			Object.keys(userProvidedScale).length === 0
 		) {
-			this.colorScale = scaleOrdinal()
-				.range(defaultColors)
-				.domain(this.allDataGroups);
-
 			return;
 		}
 
 		/**
 		 * Go through allDataGroups. If a data group has a color value provided
 		 * by the user, add that to the color range
-		 * If not, add a default color
 		 */
 		const colorRange = [];
-		let colorIndex = 0;
 		this.allDataGroups.forEach((dataGroup) => {
 			if (userProvidedScale[dataGroup]) {
 				colorRange.push(userProvidedScale[dataGroup]);
-			} else {
-				colorRange.push(defaultColors[colorIndex]);
-			}
-
-			if (colorIndex === defaultColors.length - 1) {
-				colorIndex = 0;
-			} else {
-				colorIndex++;
 			}
 		});
 
 		this.colorScale = scaleOrdinal()
 			.range(colorRange)
 			.domain(this.allDataGroups);
 	}
+
+	/* 
+	 * Color palette
+	 */
+	protected setColorClasses() {
+		const options = this.getOptions();
+		const userProvidedScale = Tools.getProperty(options, "color", "scale");
+
+		let paletteIndex = Tools.getProperty(
+			options,
+			"color",
+			"presetPalette",
+			"index"
+		);
+		const paletteAvailableChoices = {
+			"1-color": 4,
+			"2-color": 5,
+			"3-color": 5,
+			"4-color": 3,
+			"5-color": 2,
+			"14-color": 1
+		};
+		const numberOfDataGroups =
+			this.allDataGroups.length > 5 ? 14 : this.allDataGroups.length;
```

Comment:
why would we use a fake `numberOfDataGroups` value if more than 5? do we need a comment here to describe the decision?